src/Centreon/Domain/ServiceConfiguration/ServiceConfigurationService.php

```diff
@@ -22,33 +22,216 @@
+                    }
+                }
+            };
+        $extractHostTemplateIdsFromHost($host, $hostTemplateIds);
+
+        /**
+         * First, we will search for services already associated with the host to avoid creating a new one with
+         * same service description.
+         */
+        $serviceAlreadyExists = $this->findServicesByHost($host);
+
+        /**
+         * Then, we memorize the alias of service.
+         * The service description is based on the alias of service template when it was created.
+         */
+        $serviceAliasAlreadyUsed = [];
+        foreach ($serviceAlreadyExists as $service) {
+            $serviceAliasAlreadyUsed[] = $service->getDescription();
+        }
+
+        /**
+         * Then, we will search for all service templates associated with the host templates
+         */
+        $hostTemplateServices = $this->findHostTemplateServices($hostTemplateIds);
+
+        /**
+         * Extract service templates associated to host template.
+         *
+         * **We retrieve service templates only from an enabled host template.**
+         *
+         * @param int $hostTemplateId Host template id for which we want to find the services templates
+         * @return Service[]
+         */
+        $extractServiceTemplatesByHostTemplate = function (int $hostTemplateId) use ($hostTemplateServices): array {
+            $serviceTemplates = [];
+            foreach ($hostTemplateServices as $hostTemplateService) {
+                if ($hostTemplateService->getHostTemplate()->getId() === $hostTemplateId) {
+                    // Only if the host template is activated
+                    if ($hostTemplateService->getHostTemplate()->isActivated()) {
+                        $serviceTemplates[] = $hostTemplateService->getServiceTemplate();
+                    }
+                }
+            }
+            return $serviceTemplates;
+        };
+
+        $servicesToBeCreated = [];
+
+        /**
+         * Then, we set aside the services to be created.
+         * We must not have two services with the same description (alias of the service template).
+         * The priority order is defined by the list of host templates.
+         * We only retrieve the service templates that are activated.
+         */
+        if (!empty($hostTemplateIds)) {
+            foreach ($hostTemplateIds as $hostTemplateId) {
+                $serviceTemplates = $extractServiceTemplatesByHostTemplate($hostTemplateId);
+
+                foreach ($serviceTemplates as $serviceTemplate) {
+                    if (!$serviceTemplate->isActivated()) {
+                        continue;
+                    }
+
+                    $illegalObjectNameCharacters = $engineConfiguration->getIllegalObjectNameCharacters();
+
+                    if (
+                        $serviceTemplate->getAlias() !== null
+                        && $illegalObjectNameCharacters !== null
```

Comment:
If the field is empty, the query will return null : 
![image](https://user-images.githubusercontent.com/34628915/91298380-39e03500-e7a0-11ea-981c-1467e9ea8620.png)
![image](https://user-images.githubusercontent.com/34628915/91298415-49f81480-e7a0-11ea-83f5-c2e3d2b3c345.png)
This mean that, the $servicesToBeCreated, will remain empty